baseplate/core.py

```diff
@@ -135,6 +136,62 @@ def from_upstream(cls, trace_id, parent_id, span_id, sampled, flags):
 
         return cls(trace_id, parent_id, span_id, sampled, flags)
 
+    @classmethod
+    def extract_upstream_header_values(cls, upstream_header_names, headers):
```

Comment:
Would it make sense just to merge this into `from_upstream` ? I don't think we'd ever use one without the other?

Reply:
Yeah... I wanted to do that but there are subtle differences in how the two protocols handle the header values due to Thrift operating in bytes and HTTP using strings.

https://github.com/reddit/baseplate/pull/188/files#diff-db4cf04c08224f72cb9627e1f9866bbcR121
https://github.com/reddit/baseplate/pull/188/files#diff-d1382533c32ca95376190e31679acc05R162

Some type checking could be done. Or possibly a generic

```
True if extraced_values["sampled"] in ("1", b"1")
```

but this started to feel protocol specific again. If others are added, other protocol specific edge cases could arise making the ``from_upstream`` method kind of gnarly. I'm happy to take a stab at it though, if you'd like.